builder/src/tarball.rs

```diff
@@ -78,10 +87,20 @@ impl Read for TarReader {
 
 impl Seek for TarReader {
     fn seek(&mut self, pos: SeekFrom) -> std::io::Result<u64> {
+        let seekable = self.seekable();
         match self {
-            TarReader::File(f) => f.seek(pos),
-            TarReader::BufReaderInfoSeekable(b) => b.seek(pos),
-            _ => Err(enosys!("seek() not supported!")),
+            TarReader::File(f) => {
+                debug_assert!(seekable);
```

Comment:
We do not recommend using `debug_assert!`, as this may cause inconsistent binary behavior between debug and release builds.

Reply:
fixed, use assert!